src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -1086,26 +1086,24 @@ public override bool HandleCall (MethodBody callingMethodBody, MethodReference c
 							_ => null,
 						};
 
+						// Otherwise fall back to the bitfield requirements
+						var requiredMemberTypes = GetDynamicallyAccessedMemberTypesFromBindingFlagsForConstructors (bindingFlags);
+						// We can scope down the public constructors requirement if we know the number of parameters is 0
+						if (requiredMemberTypes == DynamicallyAccessedMemberTypes.PublicConstructors && ctorParameterCount == 0)
+							requiredMemberTypes = DynamicallyAccessedMemberTypes.PublicParameterlessConstructor;
```

Comment:
We calculate always the requiredMemberTypes but, for most operations, we end up not using it and we just use the binding flags. I don't see the advantage of having it up here.